### PR TITLE
lib: CURL_WRITEFUNC_ABORT is a new write callback error

### PR DESCRIPTION
**Pull request by bagder**
... which makes libcurl return CURLE_ABORTED_BY_CALLBACK as error instead of CURLE_WRITE_ERROR.

Ref: #12201

### CODE REVIEW

File: docs/libcurl/opts/CURLOPT_WRITEFUNCTION.3 [hunk 1]
```diff
@@ -63,8 +63,11 @@ that amount differs from the amount passed to your callback function, it
 signals an error condition to the library. This causes the transfer to get
 aborted and the libcurl function used returns \fICURLE_WRITE_ERROR\fP.
 
-You can also abort the transfer by returning CURL_WRITEFUNC_ERROR (added in
-7.87.0), which makes \fICURLE_WRITE_ERROR\fP get returned.
+You can stop the transfer by returning CURL_WRITEFUNC_ERROR (added in 7.87.0),
+which makes \fICURLE_WRITE_ERROR\fP get returned.
+
+You can abort the transfer by returning CURL_WRITEFUNC_ABORT (added in 8.5.0),
```
Comment (jay) [inline docs/libcurl/opts/CURLOPT_WRITEFUNCTION.3:69]: i think users may find this confusing. if stopping and aborting are the same thing except for the error code then i would refer to both of them one way or the other

**Comment by bagder** [comment]
I'm having seconds thoughts on this PR. Does this really add enough value?

**Comment by jay** [comment]
> I'm having seconds thoughts on this PR. Does this really add enough value?

As I understand it the user that requested this feature wants to be able to distinguish between a callback error and a write error caused by any other libcurl function. In practice I doubt that's much of an issue, judging by the results of `git grep CURLE_WRITE_ERROR lib`. OTOH I like it for consistency. The read function has CURL_READFUNC_ABORT => CURLE_ABORTED_BY_CALLBACK and so it makes sense to have CURL_WRITEFUNC_ABORT => CURLE_ABORTED_BY_CALLBACK.

I tagged this as needs-votes and we can wait for more feedback.

**Comment by bagder** [comment]
No one has voted in favor, and I think most users already provide their own error reason in a private struct field already if they want one. I think this PR only makes adds new code path combinations for very little extra gain or use.

**Pull request closed without merging**